hack/functests.sh

```diff
@@ -37,10 +35,48 @@ if [[ ${KUBEVIRT_PROVIDER} == os-* ]] || [[ ${KUBEVIRT_PROVIDER} =~ (okd|ocp)-*
     oc=${kubectl}
 fi
 
-test_options="-kubeconfig=${kubeconfig} -container-tag=${docker_tag} -container-tag-alt=${docker_tag_alt} -container-prefix=${functest_docker_prefix} -image-prefix-alt=${image_prefix_alt} -oc-path=${oc} -kubectl-path=${kubectl} -gocli-path=${gocli} -test.timeout 420m ${FUNC_TEST_ARGS} -installed-namespace=${namespace} -previous-release-tag=${PREVIOUS_RELEASE_TAG} -previous-release-registry=${previous_release_registry} -deploy-testing-infra=${deploy_testing_infra}"
+rm -rf $ARTIFACTS
+mkdir -p $ARTIFACTS
+
+function functest() {
+    _out/tests/ginkgo -r --slowSpecThreshold 60 $@ _out/tests/tests.test -- -kubeconfig=${kubeconfig} -container-tag=${docker_tag} -container-tag-alt=${docker_tag_alt} -container-prefix=${functest_docker_prefix} -image-prefix-alt=${image_prefix_alt} -oc-path=${oc} -kubectl-path=${kubectl} -gocli-path=${gocli} -installed-namespace=${namespace} -previous-release-tag=${PREVIOUS_RELEASE_TAG} -previous-release-registry=${previous_release_registry} -deploy-testing-infra=${deploy_testing_infra} -config=${KUBEVIRT_DIR}/tests/default-config.json --artifacts=${ARTIFACTS}
+}
+
+set -x
 if [[ ${KUBEVIRT_PROVIDER} =~ .*(k8s-1\.16)|(k8s-1\.17)|k8s-sriov.* ]]; then
     echo "Will skip test asserting the cluster is in dual-stack mode."
-    test_options="$test_options -skip-dual-stack-test"
+    FUNC_TEST_ARGS="${FUNC_TEST_ARGS} -skip-dual-stack-test"
 fi
 
-${TESTS_OUT_DIR}/tests.test ${test_options}
+if [ "$KUBEVIRT_E2E_PARALLEL" == "true" ]; then
+    parallel_test_args=""
+    serial_test_args=""
+
+    if [ -n "$KUBEVIRT_E2E_SKIP" ]; then
+        parallel_test_args="${parallel_test_args} --skip=\\[Serial\\]|${KUBEVIRT_E2E_SKIP}"
+        serial_test_args="${serial_test_args} --skip=${KUBEVIRT_E2E_SKIP}"
+    else
+        parallel_test_args="${parallel_test_args} --skip=\\[Serial\\]"
+    fi
+
+    if [ -n "$KUBEVIRT_E2E_FOCUS" ]; then
+        parallel_test_args="${parallel_test_args} --focus=${KUBEVIRT_E2E_FOCUS}"
+        serial_test_args="${serial_test_args} --focus=\\[Serial\\].*(${KUBEVIRT_E2E_FOCUS})|(${KUBEVIRT_E2E_FOCUS}).*\\[Serial\\]"
+    else
+        serial_test_args="${serial_test_args} --focus=\\[Serial\\]"
+    fi
+
+    functest --nodes=3 ${parallel_test_args} ${FUNC_TEST_ARGS}
```

Comment:
Maybe we should make this into an env var `KUBEVIRT_E2E_PARALLEL_NODES` so we can adjust from the prow job config?